lms/djangoapps/teams/views.py

```diff
@@ -1167,6 +1169,12 @@ def post(self, request):
                 status=status.HTTP_400_BAD_REQUEST
             )
 
+        if is_instructor_managed_team(team) and not _has_course_staff_privileges(request.user, team.course_id):
+            return Response(
+                build_api_error(ugettext_noop("You can't join an instructor managed team.")),
+                status=status.HTTP_400_BAD_REQUEST
```

Comment:
Let's return 403 here instead, as the API request is valid but the user is not allowed to create the membership.